crates/storage/provider/src/providers/static_file/manager.rs

```diff
@@ -1364,13 +1364,13 @@ impl TransactionsProviderExt for StaticFileProvider {
         // chunks are too big, there will be idle threads waiting for work. Choosing an
         // arbitrary smaller value to make sure it doesn't happen.
         let chunk_size = 100;
-        let mut channels = Vec::new();
 
         // iterator over the chunks
         let chunks = tx_range
             .clone()
             .step_by(chunk_size)
             .map(|start| start..std::cmp::min(start + chunk_size as u64, tx_range.end));
+        let mut channels = Vec::with_capacity(chunks.clone().count());
```

Comment:
you can probably do smthg like `tx_range.end() - tx.start()` here instead of cloning iterator to count